surface/include/pcl/surface/mls.h

```diff
@@ -283,6 +306,18 @@ namespace pcl
       inline int
       getDilationIterations () { return dilation_iteration_num_; }
 
+      /** /brief Set wether the mls results should be stored for each point in the input cloud
+       *  /param[in] True if the mls results should be stored, otherwise false.
+       */
+      inline void
+      setCacheMLSResults(bool cache_mls_results) {cache_mls_results_ = cache_mls_results;}
+
+      /** /brief Get the MLSResults for input cloud
+       *  /note The results are only stored if setCacheMLSResults(true) was called
+       *        or when using the upsampling method DISTINCT_CLOUD or VOXEL_GRID_DILATION.
+       */
+      inline std::vector<MLSResult> getMLSResults() const {return mls_results_;}
```

Comment:
Why not returning by const ref? Otherwise will be quite a costly function.
Also missing spaces as above.